export_nonpublic.patch

```diff
@@ -0,0 +1,20 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 8581b4c..d81cbec 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -717,6 +717,15 @@ install(FILES ${CERES_HDRS} DESTINATION include/ceres)
+ file(GLOB CERES_PUBLIC_INTERNAL_HDRS ${CMAKE_SOURCE_DIR}/include/ceres/internal/*.h)
+ install(FILES ${CERES_PUBLIC_INTERNAL_HDRS} DESTINATION include/ceres/internal)
+ 
++# Optionally install the non-public-headers.
++if (EXPORT_NONPUBLIC_HEADERS)
++  message("-- Export non-public headears.")
++  file(GLOB CERES_NON_PUBLIC_HDRS ${CMAKE_SOURCE_DIR}/internal/ceres/*.h)
++  install(FILES ${CERES_NON_PUBLIC_HDRS} DESTINATION include/ceres)
```

Comment:
That might not work, as the (internal) headers rely on this path structure for their includes.